src/popup/components/Modals/PrivateKeyImport.vue

```diff
@@ -0,0 +1,159 @@
+
+import type { Protocol, ResolveCallback } from '@/types';
+
+import { ProtocolAdapterFactory } from '@/lib/ProtocolAdapterFactory';
+import { ACCOUNT_TYPES, MODAL_HELP } from '@/constants';
+import { useAccounts, useModals } from '@/composables';
+
+import BtnMain from '@/popup/components/buttons/BtnMain.vue';
+import FormTextarea from '@/popup/components/form/FormTextarea.vue';
+import Modal from '@/popup/components/Modal.vue';
+
+export default defineComponent({
+  components: {
+    BtnMain,
+    Modal,
+    FormTextarea,
+  },
+  props: {
+    resolve: { type: Function as PropType<ResolveCallback>, required: true },
+    protocol: { type: String as PropType<Protocol>, required: true },
+  },
+  setup(props) {
+    const { t } = useI18n();
+
+    const { accounts, addPrivateKeyAccount, setActiveAccountByGlobalIdx } = useAccounts();
+    const { openModal } = useModals();
+
+    const privateKey = ref('');
+    const error = ref<TranslateResult>('');
+
+    function getProtocolName(protocol: Protocol) {
+      return ProtocolAdapterFactory.getAdapter(protocol).protocolName;
+    }
+
+    async function importPrivateKey() {
+      const adapter = ProtocolAdapterFactory.getAdapter(props.protocol);
+      try {
+        const account = adapter.resolveAccountRaw(
+          {
+            type: ACCOUNT_TYPES.privateKey,
+            isRestored: false,
+            protocol: props.protocol,
+            privateKey: Buffer.from(privateKey.value, 'hex'),
+          },
+          0,
+          0,
+        );
+
+        if (!account) {
+          throw new Error('');
+        }
+
+        if (
+          accounts.value.find(({ protocol, secretKey }) => (
+            Buffer.from(secretKey).toString('hex') === Buffer.from(account.secretKey).toString('hex')
+            && protocol === props.protocol
+          ))
+        ) {
+          error.value = t('modals.privateKeyImport.privateKeyExist');
+          return;
+        }
+      } catch (e) {
+        error.value = t('modals.privateKeyImport.invalidPrivateKey', {
+          protocol: getProtocolName(props.protocol),
+        });
+        return;
+      }
+      const globalIdx = addPrivateKeyAccount({
+        type: ACCOUNT_TYPES.privateKey,
+        isRestored: false,
+        protocol: props.protocol,
+        privateKey: Buffer.from(privateKey.value, 'hex'),
+      });
+      setActiveAccountByGlobalIdx(globalIdx);
+      props.resolve();
+    }
+
+    function showPrivateKeyHelp() {
+      openModal(MODAL_HELP, {
+        title: t('modals.privateKeyImport.helpTitle'),
+        msg: t('modals.privateKeyImport.helpMsg'),
+        textCenter: true,
+      });
+    }
+
+    watch(privateKey, () => {
+      error.value = '';
+    });
+
+    return {
+      error,
+      privateKey,
+      getProtocolName,
+      importPrivateKey,
+      showPrivateKeyHelp,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+@use '@/styles/typography';
+
+.private-key-import .text-center {
```

Comment:
I strongly suggest to avoid applying additional styles to utility classes defined in `global.scss` file (`text-center`). I suggest to add new class to apply some custom styling.